crates/node/src/roles/base.rs

```diff
@@ -37,52 +37,39 @@ impl BaseRole {
     }
 
     pub fn start(self) -> Result<(), ShutdownError> {
-        TaskCenter::spawn_child(TaskKind::RoleRunner, "base-role-service", async {
-            let cancelled = cancellation_watcher();
-
-            tokio::select! {
-                result = self.run() => {
-                    result
-                }
-                _ = cancelled =>{
-                    Ok(())
-                }
-            }
+        // We don't want this task to shutdown by task center, we intentionally want to leak it and
+        // leave it running until the runtime is terminated.
+        //
+        // This is because we want to try and response with NodeStatus::ShuttingDown for as long as
```

Comment:
```suggestion
        // This is because we want to respond while NodeStatus::ShuttingDown for as long as
```
?

Reply:
This actually is old comment. will remove.